### PR TITLE
C# UWP WinRT sample application for using a Windows Studio camera and its set of effects

### PR DESCRIPTION
**Pull request by LPBourret**
We had request from partners to develop a sample completely in a single C# project (without component dependencies or C++WinRT) to showcase how to be aware of the presence of a camera opted into Windows Studio as well as how to interact with its set of effects via generic DDI SET and GET APIs (some of which we already covered using C++WinRT).

### CODE REVIEW

File: Samples/WindowsStudio/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        // set the flags value for the corresponding extended control
+        KsHelper.SetExtendedControlFlags(m_mediaCapture.VideoDeviceController, KsHelper.ExtendedControlKind.KSPROPERTY_CAMERACONTROL_EXTENDED_EYEGAZECORRECTION, flagToSet); 
+        ```
+## Requirements
+	
+This sample is built using Visual Studio 2019 and requires [Windows SDK version 22621](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/) at the very least.
+
+## Using the samples
+
+The easiest way to use these samples without using Git is to download the zip file containing the current version (using the following link or by clicking the "Download ZIP" button on the repo page). You can then unzip the entire archive and use the samples in Visual Studio 2017.
```
Comment (saredd) [inline Samples/WindowsStudio/README.md:59]: Is this supposed to be 2019?

Comment (saredd) [reply Samples/WindowsStudio/README.md:59]: Same below

File: Samples/WindowsStudio/UWPWindowsStudioSample/UWPWindowsStudioSample.sln [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "UWPWindowsStudioSample", "UWPWindowsStudioSample.csproj", "{C1544F26-64CB-432F-8CEF-CA23412F2D9D}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|ARM = Debug|ARM
+		Debug|ARM64 = Debug|ARM64
+		Debug|x64 = Debug|x64
+		Debug|x86 = Debug|x86
+		Release|ARM = Release|ARM
+		Release|ARM64 = Release|ARM64
```
Comment (NaveenThum) [inline Samples/WindowsStudio/UWPWindowsStudioSample/UWPWindowsStudioSample.sln:15]: 32-bit ARM is not supported any longer.

File: Samples/WindowsStudio/UWPWindowsStudioSample/UWPWindowsStudioSample.csproj [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    <NoWarn>;2008</NoWarn>
+    <DebugType>pdbonly</DebugType>
+    <PlatformTarget>x86</PlatformTarget>
+    <UseVSHostingProcess>false</UseVSHostingProcess>
+    <ErrorReport>prompt</ErrorReport>
+    <Prefer32Bit>true</Prefer32Bit>
+    <UseDotNetNativeToolchain>true</UseDotNetNativeToolchain>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|ARM'">
+    <DebugSymbols>true</DebugSymbols>
```
Comment (NaveenThum) [inline Samples/WindowsStudio/UWPWindowsStudioSample/UWPWindowsStudioSample.csproj:46]: Drop 32-bit ARM

File: Samples/WindowsStudio/UWPWindowsStudioSample/KsHelper.cs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            var size = Marshal.SizeOf<T>();
+            var intPtr = Marshal.AllocHGlobal(size);
+            var bytes = new byte[size];
+            Marshal.StructureToPtr<T>(item, intPtr, false);
+            Marshal.Copy(intPtr, bytes, 0, size);
+            Marshal.FreeHGlobal(intPtr);
+            return bytes;
+        }
+
+        // A clunky method to marshal a byte[] into a structure
```
Comment (NaveenThum) [inline Samples/WindowsStudio/UWPWindowsStudioSample/KsHelper.cs:199]: >clunky [](http://example.com/codeflow?start=13&length=6)

clunky or clever?

File: Samples/WindowsStudio/UWPWindowsStudioSample/KsHelper.cs [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+            // send a SET command with the KsProperty and payload and retrieve the result status
+            var resultStatus = controller.SetDevicePropertyByExtendedId(byteKsProp, bytePayload);
+            
+            if (resultStatus != VideoDeviceControllerSetDevicePropertyStatus.Success)
+            {
+                throw new Exception($"Unexpectedly could not SET flags={flags} for control={controlKind}, status={resultStatus}");
+            }
+        }
+
+        // A clunky method to marshal a structure into a byte[]
```
Comment (NaveenThum) [inline Samples/WindowsStudio/UWPWindowsStudioSample/KsHelper.cs:187]: clunky or clever?

**Comment by NaveenThum** [comment]
> using System;

Copyright?

---
Refers to: Samples/WindowsStudio/UWPWindowsStudioSample/App.xaml.cs:1 in 550d07e. [](commit_id = 550d07e66d5b682631741ac5afd56de276fb994e, deletion_comment = False)

**Review by NaveenThum — APPROVED** [review]
:shipit:

**Pull request merged**